detectron2/modeling/poolers.py

```diff
@@ -83,10 +90,13 @@ def convert_boxes_to_pooler_format(box_lists: List[Boxes]):
     """
     boxes = torch.cat([x.tensor for x in box_lists], dim=0)
     # __len__ returns Tensor in tracing.
-    sizes = shapes_to_tensor([x.__len__() for x in box_lists], device=boxes.device)
+    sizes = shapes_to_tensor([x.__len__() for x in box_lists])
+    sizes = move_tensor_device_same_as_another(sizes, boxes)
     indices = torch.repeat_interleave(
-        torch.arange(len(box_lists), dtype=boxes.dtype, device=boxes.device), sizes
+        move_tensor_device_same_as_another(torch.arange(len(box_lists), dtype=boxes.dtype), boxes),
```

Comment:
Code like this might still have performance hit or maybe don't even run at all, because after tracing it will be translated to
```
a = torch.arange(..., device='old_device').to(device='new_device')
# instead of a = torch.arange(device='new_device')
```
If old_device=cuda0 and new_device=cuda1, this line triggers a gpu-gpu copy that's unnecessary.
If old_device=cuda and new_device=cpu, the traced model doesn't work on a pure-CPU machine at all.

My hacky `ts_to_device` script probably works for this one.